engine/src/main/java/org/terasology/engine/logic/common/RetainComponentsComponent.java

```diff
@@ -15,7 +15,12 @@
  * If a block (item) entity has a component that is not part of its prefab, retaining this component results in the block
  * entity still having this component afterwards. If not retained, it is likely to be removed instead.
  */
-public class RetainComponentsComponent implements Component {
+public class RetainComponentsComponent implements Component<RetainComponentsComponent> {
     @Replicate
     public Set<Class<? extends Component>> components = Sets.newHashSet();
+
+    @Override
+    public void copy(RetainComponentsComponent other) {
+        this.components = Sets.newHashSet(other.components); // TODO Investigate, needs to deep-copy or not!
```

Comment:
probably not? we're on the meta-level here and just reference component classes not component instances.